File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/ValueSet.ts
import { UsageContext } from "./UsageContext";
import { ContactDetail } from "./ContactDetail";
import { CodeableConcept } from "./CodeableConcept";
import { Coding } from "./Coding";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type ValueSetComposeIncludeFilter = BackboneElement & {
    op: string;
    value: string;
    property: string;
};

export type ValueSetComposeIncludeConceptDesignation = BackboneElement & {
    use?: Coding;
    value: string;
    language?: string;
};

export type ValueSetComposeIncludeConcept = BackboneElement & {
    code: string;
    display?: string;
    designation?: ValueSetComposeIncludeConceptDesignation[];
};

export type ValueSetComposeInclude = BackboneElement & {
    filter?: ValueSetComposeIncludeFilter[];
    system?: string;
    concept?: ValueSetComposeIncludeConcept[];
    version?: string;
    valueSet?: string[];
};

export type ValueSetCompose = BackboneElement & {
    exclude?: Reference[];
    include: ValueSetComposeInclude[];
    inactive?: boolean;
    lockedDate?: string;
};

export type ValueSetExpansionContains = BackboneElement & {
    code?: string;
    system?: string;
    display?: string;
    version?: string;
    abstract?: boolean;
    contains?: Reference[];
    inactive?: boolean;
    designation?: Reference[];
};

export type ValueSetExpansionParameter = BackboneElement & {
    name: string;
    value?: string | number | boolean;
};

export type ValueSetExpansion = BackboneElement & {
    total?: number;
    offset?: number;
    contains?: ValueSetExpansionContains[];
    parameter?: ValueSetExpansionParameter[];
    timestamp: string;
    identifier?: string;
};

export type ValueSet = DomainResource & {
    description?: string;
    compose?: ValueSetCompose;
    date?: string;
    publisher?: string;
    jurisdiction?: CodeableConcept[];
    purpose?: string;
    name?: string;
    useContext?: UsageContext[];
    copyright?: string;
    experimental?: boolean;
    expansion?: ValueSetExpansion;
    title?: string;
    status: string;
    url?: string;
    identifier?: Identifier[];
    immutable?: boolean;
    version?: string;
    contact?: ContactDetail[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/ValueSet.py
from typing import Optional, List
from pydantic import *
from ..base import *

class ValueSet_Compose_Include_Filter(BackboneElement):
    op: str
    value: str
    property: str

class ValueSet_Compose_Include_Concept_Designation(BackboneElement):
    use: Optional[Coding] = None
    value: str
    language: Optional[str] = None

class ValueSet_Compose_Include_Concept(BackboneElement):
    code: str
    display: Optional[str] = None
    designation: Optional[List[ValueSet_Compose_Include_Concept_Designation]] = None

class ValueSet_Compose_Include(BackboneElement):
    filter: Optional[List[ValueSet_Compose_Include_Filter]] = None
    system: Optional[str] = None
    concept: Optional[List[ValueSet_Compose_Include_Concept]] = None
    version: Optional[str] = None
    value_set: Optional[List[str]] = None

class ValueSet_Compose(BackboneElement):
    exclude: Optional[List[Reference]] = None
    include: list[ValueSet_Compose_Include] = []
    inactive: Optional[bool] = None
    locked_date: Optional[str] = None

class ValueSet_Expansion_Contains(BackboneElement):
    code: Optional[str] = None
    system: Optional[str] = None
    display: Optional[str] = None
    version: Optional[str] = None
    abstract: Optional[bool] = None
    contains: Optional[List[Reference]] = None
    inactive: Optional[bool] = None
    designation: Optional[List[Reference]] = None

class ValueSet_Expansion_Parameter(BackboneElement):
    value_code: Optional[str] = None
    value_uri: Optional[str] = None
    value_decimal: Optional[float] = None
    name: str
    value_string: Optional[str] = None
    value_boolean: Optional[bool] = None
    value_date_time: Optional[str] = None
    value_integer: Optional[int] = None

class ValueSet_Expansion(BackboneElement):
    total: Optional[int] = None
    offset: Optional[int] = None
    contains: Optional[List[ValueSet_Expansion_Contains]] = None
    parameter: Optional[List[ValueSet_Expansion_Parameter]] = None
    timestamp: str
    identifier: Optional[str] = None

class ValueSet(DomainResource):
    description: Optional[str] = None
    compose: Optional[ValueSet_Compose] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    name: Optional[str] = None
    use_context: Optional[List[UsageContext]] = None
    copyright: Optional[str] = None
    experimental: Optional[bool] = None
    expansion: Optional[ValueSet_Expansion] = None
    title: Optional[str] = None
    status: str
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    immutable: Optional[bool] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None